src/IO/S3/URI.cpp

```diff
@@ -26,7 +26,7 @@ struct URIConverter
     static void modifyURI(Poco::URI & uri, std::unordered_map<std::string, std::string> mapper)
     {
         Macros macros({{"bucket", uri.getHost()}});
-        uri = macros.expand(mapper[uri.getScheme()]).empty()? uri : Poco::URI(macros.expand(mapper[uri.getScheme()]) + "/" + uri.getPathAndQuery());
+        uri = macros.expand(mapper[uri.getScheme()]).empty()? uri : Poco::URI(macros.expand(mapper[uri.getScheme()]) + (uri.getPathAndQuery()[0] == '/' ? "" : "/") + uri.getPathAndQuery());
```

Comment:
`URI` has the ctor of two parameters or maybe with the help of any of its other methods we could avoid manually dealing with slashes?

Reply:
I removed unused check and decided not do add any slashes at all (they are all in `getPathAndQuery()`), it is even simpler now